storage/innobase/xtrabackup/src/xtrabackup.cc

```diff
@@ -1995,7 +1998,7 @@ xtrabackup_print_metadata(char *buf, size_t buf_len)
 		 metadata_from_lsn,
 		 metadata_to_lsn,
 		 metadata_last_lsn,
-		 MY_TEST(xtrabackup_compact == TRUE),
+		 xtrabackup_compact ? (xtrabackup_prepare ? 2 : 1) : 0,
```

Comment:
This implicitly controls the value of `xtrabackup_compact_need_expand` on next read in `xtrabackup_read_metadata()` please add a comment here why.